filetree/CMakeLists.txt

```diff
@@ -50,12 +52,13 @@ new_ec_test(root_file_mixed_case root_file.in root_mixed/test.a "^child=true[ \t
 new_ec_test(root_pattern root_file.in root "^name=root[ \t\n\r]*$")
 
 # Tests path separator match
-new_ec_test(path_separator path_separator.in path/separator "^key=value[ \t\n\r]*$")
+new_ec_test(path_separator path_separator.in path/separator "^key1=value1[ \t\n\r]*$")
 
 # Windows style path separator in the command line should work on Windows, but
 # should not work on other systems
+# TODO figure out how to handle Cygwin
```

Comment:
Can Cygwin automatically recognize backslashes as slashes?

Reply:
No, it can't. Cygwin is a POSIX-compatible layer, so it expects slashes. That is what `cygpath` is for. To convert Windows paths to Cygwin paths, including the right mountpoint.

The is a very similar question at the Cygwin mailing list at the moment: https://www.cygwin.com/ml/cygwin/2018-11/msg00141.html

Reply:
I am also not aware of a way.  CMake does not define WIN32 on Cygwin, so this conditional skips the test on Cygwin. 

@ffes Thanks for the link.  That makes me think that perhaps I should just remove the "TODO" line.

Reply:
Perhaps we should just skip the related tests :)

Reply:
@xuhdev We certainly could, and I am happy to make that edit.  However, the test is meaningful, to the extent that this is actually a difference in platform-specific behaviour.  I propose leaving the test, but removing the TODO line, and adding `cmake_minimum_required` so we are guaranteed that Cygwin doesn't set `WIN32` ([CMake commit reference](https://gitlab.kitware.com/cmake/cmake/commit/85c0a69a92e78275ea0b180482bafcdb877b0dc3)).  What do you think?

Reply:
I looked at it again. Yes, I agree with you that the test is meaningful. If cygwin does not specifically handle backslashes in paths, then in principle I think we should treat it like a UNIX system, i.e., backslash is backslash.

Reply:
Sure. Let's enforce 3.5

Reply:
@xuhdev Thanks!  I updated that, and rebased on the latest master.